src/neo4j_genai/experimental/pipeline/pipeline.py

```diff
@@ -428,21 +482,50 @@ def connect(
             )
         if self.is_cyclic():
             raise PipelineDefinitionError("Cyclic graph are not allowed")
+        # invalidate the pipeline if it was already validated
+        self.is_validated = False
+
+    def validate_connection_parameters(self) -> None:
```

Comment:
I'm wondering if this is the best name for this method. If this method just validates that the parameters passed between components internally in the pipeline are correct maybe something like `validate_internal_conn_params` might be better? The method below could be `validate_all_conn_params` then. I worry `validate_connection_parameters` makes a reader think the inputs to the pipeline are also being validated here.

Reply:
Wdyt about `validate_parameter_mapping`? The parameter definition is defined by the user, it's not purely internal.
Also, there is no user input to this method, so it can't validate them, but the docstring is misleading, will be fixed.

Reply:
I meant this method doesn't validate the inputs which are provided to components when running the pipeline (if I'm understanding it correctly) whereas `validate_all_parameters` does do this. Maybe all that's needed is to update the docstring a bit here to make it clear the difference between the 2 methods.

Reply:
Yes that's exactly what these methods are doing. Potentially improved naming [here](https://github.com/neo4j/neo4j-genai-python/pull/122/commits/4494799b5724282f74f4ad45abb00beae5aa1a28). If you think it's still confusing, we can go with your first proposal.